### ISSUE TITLE
variable name with dot in callVariable returning text before dot.

### ISSUE THREAD

**Issue by aubrijesh** [handsontable/formula-parser#60]
Hi,
I am trying to use variable name with dot as I am trying to nested object values. When I am calling function with variable name with dot then in callVariable I am getting only name before dot.
example:
fmParser.parse('CONCATENATE(name.firstname, name.lastname)');

fmParser.on('callVariable', function(name, done) {
    console.log("call variable is called", name); // logging as name...
}
I am expecting that name will be name.firstname and name.lastname in callVariable function.

**Comment by budnix** [comment]
Yep, this can be treated as a limitation. Currently, grammar parser allows a dot in the formula names, not in the variable names (https://github.com/handsontable/formula-parser/blob/develop/src/grammar-parser/grammar-parser.jison#L15). 

I can mark this issue as a feature request. Although it has to be investigated if this won't break any existing feature.

As a workaround, I can suggest to you to using underscores instead of dots and convert them to a nested object.
```
fmParser.parse('CONCATENATE(name_firstname, name_lastname)');

fmParser.on('callVariable', function(name, done) {
  const name = name.replace('_', '.');
  console.log("call variable is called", name); // logging as name...
}
```

**Comment by aubrijesh** [comment]
Thanks @budnix. Right now my variable names may contains underscore so I will need to use some other delimiter. This will be very helpful if you can add this as a feature. Is there any way I can use this feature in current build?

**Comment by budnix** [comment]
Currently, I've delegated to do other things and I'm not able to implement this feature in the nearest future. Please do not close this issue until it'll be investigated.

**Comment by sshim-cb** [comment]
Any updates on this issue?

**Comment by AMBudnik** [comment]
We are switching to a new formula engine (that took us nearly a year to develop) so we won't be maintaining this project. I encourage everyone to fork it.

**Comment by AMBudnik** [comment]
> We are switching to a new formula engine

Hi @sshim-cb @aubrijesh we've just released the new formula engine https://github.com/handsontable/hyperformula
It lacks some formulas, but it should definitely meet all performance expectations.
